src/sentry/eventstore/models.py

```diff
@@ -43,13 +51,153 @@ def data(self, value):
             node_id, data=value, wrapper=EventDict, ref_version=2, ref_func=ref_func
         )
 
+    @property
+    def group_id(self):
+        if self._group_id:
+            return self._group_id
+
+        column = self.__get_column_name(Columns.GROUP_ID)
+        if column in self._snuba_data and self._snuba_data[column]:
+            return self._snuba_data[column]
+        return None
+
+    @group_id.setter
+    def group_id(self, value):
+        self._group_id = value
+
     @property
     def platform(self):
+        column = self.__get_column_name(Columns.PLATFORM)
+        if column in self._snuba_data:
+            return self._snuba_data[column]
         return self.data.get("platform", None)
 
+    @property
+    def message(self):
+        if self._message:
+            return self._message
+
+        column = self.__get_column_name(Columns.MESSAGE)
+        if column in self._snuba_data:
+            return self._snuba_data[column]
+
+        return self.data.get("message")
+
+    @message.setter
+    def message(self, value):
+        self._message = value
+
     @property
     def datetime(self):
-        recorded_timestamp = self.data.get("timestamp")
-        date = datetime.fromtimestamp(recorded_timestamp)
-        date = date.replace(tzinfo=timezone.utc)
+        column = self.__get_column_name(Columns.TIMESTAMP)
+        if column in self._snuba_data:
+            return parse_date(self._snuba_data[column]).replace(tzinfo=pytz.utc)
+
+        timestamp = self.data.get("timestamp")
+        date = datetime.fromtimestamp(timestamp)
+        date = date.replace(tzinfo=pytz.utc)
         return date
+
+    @property
+    def timestamp(self):
+        column = self.__get_column_name(Columns.TIMESTAMP)
+        if column in self._snuba_data:
+            return self._snuba_data[column]
+        return self.datetime.isoformat()
```

Comment:
IS this here just because SnubaEvent has one ?

Reply:
Sorry, which part are you referring to?

Reply:
The existence of both datetime and timestamp

Reply:
Yes, we rely on event.timestamp in various places